tests/form_test.py

```diff
@@ -114,6 +114,40 @@ def test_init(self):
                 self.f.sys_vars["y"] == "boolean")
 
 
+def test_declare():
+    g = GRSpec()
+    g.declare('a')
+    assert g.sys_vars == dict(a='boolean'), g.sys_vars
+    assert g.env_vars == dict(), g.env_vars
+    g.declare('a', 'b')
+    assert g.sys_vars == dict(a='boolean', b='boolean'), g.sys_vars
+    assert g.env_vars == dict(), g.env_vars
+    g.declare('c', env=True)
+    assert g.sys_vars == dict(a='boolean', b='boolean'), g.sys_vars
+    assert g.env_vars == dict(c='boolean'), g.env_vars
+    # attempt to redeclare "c" as sys var
+    with nt.assert_raises(AssertionError):
+        g.declare('c')
+    g = GRSpec()
```

Comment:
I split the test into multiple functions, per type of values that the declared variables take. This should make it faster to track the program's state up to the point where an assertion has failed.